src/app/features/notification/notification.component.ts

```diff
@@ -35,80 +37,157 @@ import { closeOnlySubmenu } from "./close-only-submenu";
+    this.notificationsSubject.next(notifications);
   }
 
   /**
-   * Filters notifications based on the sender.
-   * @param notifications - The list of notifications.
-   * @param user - The user to filter notifications.
+   * Loads the user's notification configuration.
    */
-  private filterUserNotifications(notifications: NotificationEvent[]) {
-    const user = this.sessionInfo.value?.id;
+  private async loadNotificationConfig(userId: string) {
+    try {
+      return await this.entityMapper.load<NotificationConfig>(
+        NotificationConfig,
+        userId,
+      );
+    } catch (error) {
+      return null;
+    }
+  }
 
+  /**
+   * Filters notifications based on the sender and read status.
+   */
+  private filterUserNotifications(notifications: NotificationEvent[]) {
     this.allNotifications = notifications.filter(
-      (notification) => notification.sentBy === user,
+      (notification) => notification.notificationFor === this.userId,
     );
     this.unreadNotifications = notifications.filter(
       (notification) =>
-        notification.sentBy === user && !notification.readStatus,
+        notification.notificationFor === this.userId &&
+        !notification.readStatus,
     );
   }
 
-  markAllRead() {
-    // TODO: Implement the logic to mark all notifications as read.
-    Logging.log("All notifications marked as read");
+  /**
+   * Marks all notifications as read.
+   */
+  async markAllRead(): Promise<void> {
+    const unreadNotifications = this.allNotifications.filter(
+      (notification) => !notification.readStatus,
+    );
+    await this.updateReadStatusForNotifications(unreadNotifications, true);
   }
 
-  enableNotificationForUser() {
-    // TODO: Need to implement the logic so that the notification is enabled corresponding to the user.
-    Logging.log("Notification enabled");
+  /**
+   * Toggles the notification setting for the user.
+   */
+  async enableNotificationForUser(): Promise<void> {
+    // TODO: Implement the logic to called the getToken function from the NotificationService file, Once the PR #2692 merged.
+    this.hasNotificationEnabled = !this.hasNotificationEnabled;
+
+    const notificationConfig = await this.loadNotificationConfig(this.userId);
+
+    // TODO: Currently, email notification are disabled. Update this logic once the email notification feature is implemented.
+    const notificationChannels = {
+      push: this.hasNotificationEnabled,
+      email: false,
+    };
+
+    if (notificationConfig) {
+      notificationConfig.channels = notificationChannels;
+      await this.entityMapper.save(notificationConfig);
+    } else {
+      const newNotificationConfig = new NotificationConfig(this.userId);
+      newNotificationConfig.channels = notificationChannels;
+      await this.entityMapper.save(newNotificationConfig);
+    }
+
+    this.alertService.addInfo(
+      `Notification ${this.hasNotificationEnabled ? "enabled" : "disabled"}.`,
+    );
+  }
```

Comment:
Let's move this to the NotificationSettings component only and remove the menu entry for it also. Code (and maybe even UI) will be much easier if this is only in the full settings page and not also in the notification panel directly.

Reply:
I have removed this code from this component and I will implement this code in the notification setting UI PR.